srHook/SAPHanaSrCostOptMemConfig.py

```diff
@@ -0,0 +1,161 @@
+        self.tracer.info("leave shutdown hook")
+        return 0
+
+    def failover(self, hostname, storage_partition, sr_mode, **kwargs):
+        self.tracer.debug("enter failover hook; %s" % locals())
+        self.tracer.debug(self.config.toString())
+        self.tracer.info("leave failover hook")
+        return 0
+
+    def stonith(self, failingHost, **kwargs):
+        self.tracer.debug("enter stonith hook; %s" % locals())
+        self.tracer.debug(self.config.toString())
+        # e.g. stonith of params["failed_host"]
+        # e-g- set vIP active
+        self.tracer.info("leave stonith hook")
+        return 0
+
+    def preTakeover(self, isForce, **kwargs):
+        """Pre takeover hook."""
+        self.tracer.info("%s.preTakeover method called with isForce=%s" % (self.__class__.__name__, isForce))
+        if not isForce:
+            # run pre takeover code
+            # run pre-check, return != 0 in case of error => will abort takeover
+            return 0
+        else:
+            # possible force-takeover only code
+            # usually nothing to do here
+            return 0
+
+    def postTakeover(self, rc, **kwargs):
+        method = "postTakeover"
+        """Post takeover hook."""
+        self.tracer.info("{0}.{1}() method called with rc={2}".format(self.__class__.__name__, method, rc))
+        # TODO: PRIO4: Do we need to differ forced (rc=1) and normal (rc=0) takeover?
+        if rc == 0 or rc == 1:
+            # normal takeover succeeded
+            # and
+            # waiting for force takeover
+            connection = dbapi.connect(
+                key=self.userkey,
+                # address='localhost',port=dbport,user=dbuser,passwort=dbpwd,
+            )
+            cursor = connection.cursor()
+            self.tracer.info("sqlstatement: {0}".format(self.sql_set_memory))
+            cursor.execute(self.sql_set_memory)
+            self.tracer.info("sqlstatement: {0}".format(self.sql_set_preload))
+            cursor.execute(self.sql_set_preload)
+            cursor.close()
+
+            # return 0
+        # elif rc == 2:
+            # error, something went wrong
+            # return 0
+
+        self.tracer.info("leave postTakeover hook")
+        return 0
+
+    def srConnectionChanged(self, ParamDict, **kwargs):
+        self.tracer.debug("enter srConnectionChanged hook; %s" % locals())
+
+        # Access to parameters dictionary
+        # hostname = ParamDict['hostname']
+        # port = ParamDict['port']
+        # database = ParamDict['database']
+        # status = ParamDict['status']
+        # databaseStatus = ParamDict['database_status']
+        # systemStatus = ParamDict['system_status']
+        # timestamp = ParamDict['timestamp']
+        # isInSync = ParamDict['is_in_sync']
+        # reason = ParamDict['reason']
+        # siteName = ParamDict['siteName']
+
+        self.tracer.info("leave srConnectionChanged hook")
+        return 0
```

Comment:
The hook seems missing some of the latest changes from branch 'SLE-18613'

Reply:
@angelabriel I have no idea how to solve that. Because it seams that this commit came in, because I did a merge with SLE-18613 in the last days. If I now pull new changes from SLE-18613 (again) it could get worse.

Reply:
@angelabriel I guess the best is still to remove the hook with an additoinal commit and solve the issues with the hook afterwards.